analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -87,12 +87,20 @@ public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
             context.RegisterCompilationAction(c =>
                 {
                     ReadParameters(context, c);
-                    if (!IsAnalyzerEnabled ||
-                        (SkipAnalysisForTestProject && IsTestProject))
+                    if (!IsAnalyzerEnabled)
                     {
                         return;
                     }
 
+                    // The results of Metrics and CopyPasteToken analyzers are not needed for Test projects yet the plugin side expects the protobuf files, so we create empty ones.
+                    if (SkipAnalysisForTestProject && IsTestProject)
```

Comment:
I'd swap the order to first define the context and then the special condition on it.
```suggestion
                    if (IsTestProject && SkipAnalysisForTestProject)
```